src/monodroid/jni/debug-app-helper.cc

```diff
@@ -200,26 +200,25 @@ static char*
 get_libmonosgen_path ()
 {
 	char *libmonoso;
-	bool embedded_dso_mode_enabled = androidSystem.is_embedded_dso_mode_enabled ();
 
 	// Android 5 includes some restrictions on loading dynamic libraries via dlopen() from
 	// external storage locations so we need to file copy the shared object to an internal
 	// storage location before loading it.
 	copy_native_libraries_to_internal_location ();
 
-	if (!embedded_dso_mode_enabled) {
-		for (size_t i = 0; i < BasicAndroidSystem::MAX_OVERRIDES; ++i) {
-			if (runtime_exists (BasicAndroidSystem::override_dirs [i], libmonoso)) {
-				return libmonoso;
-			}
+	if (androidSystem.is_embedded_dso_mode_enabled ()) {
+		return (char*) SharedConstants::MONO_SGEN_SO;
```

Comment:
I don't actually know if this cast is right.

Reply:
It's not needed - the constant has the `char[]` type, which is the same as `char*`

Reply:
If I remove it, I get:
```
debug-app-helper.cc(210,10): error GF586CA7D: cannot initialize return object of type 'char *' with an lvalue of type 'const char [19]'
```

Reply:
ugh, const, right. Change the return value of the function to `const char*` (on lines 31 and 90 too), that should do it